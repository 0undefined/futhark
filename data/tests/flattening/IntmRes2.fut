fun [int] addToRow ([int] xs, int y) =
  map(fn int (int x) => x+y, xs)

fun [[[int]]] main ([[[int]]] xsss, [int] cs, int y) =
  map (fn [[int]] ([[int]] xss, int c) =>
         let y' = y * c + c in
         map(fn [int] ([int] xs) =>
               addToRow(xs,y')
            , xss)
      , zip (xsss,cs))
