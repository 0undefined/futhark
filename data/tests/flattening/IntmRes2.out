[[[7, 8, 9],
  [10, 11, 12]],
 [[18, 19, 20],
  [21, 22, 23]],
 [[21, 20, 19],
  [22, 23, 24]],
 [[32, 31, 30],
  [35, 34, 33]]]
